jetty-core/jetty-quic/jetty-quic-api/src/main/java/org/eclipse/jetty/quic/api/Stream.java

```diff
@@ -0,0 +1,428 @@
+         *
+         * @param stream the stream
+         * @see Stream#demand()
+         */
+        default void onDataAvailable(Stream stream)
+        {
+            while (true)
+            {
+                Data data = stream.read();
+                if (data == null)
+                {
+                    stream.demand();
+                    return;
+                }
+                data.release();
+                if (data.isLast())
+                    return;
+            }
+        }
+
+        /**
+         * <p>Invoked when a STREAM_DATA_BLOCKED frame has been received.</p>
+         * <p>This event is only emitted for informational purposes.</p>
+         *
+         * @param stream the stream
+         * @param frame the frame
+         */
+        default void onDataBlocked(Stream stream, StreamDataBlockedFrame frame)
+        {
+        }
+
+        /**
+         * <p>Invoked when a MAX_STREAM_DATA frame has been received.</p>
+         * <p>This event is only emitted for informational purposes.</p>
+         *
+         * @param stream the stream
+         * @param frame the frame
+         */
+        default void onMaxData(Stream stream, StreamMaxDataFrame frame)
+        {
+        }
+
+        /**
+         * <p>Invoked when a STOP_SENDING frame has been received.</p>
+         * <p>This event is only emitted for informational purposes.</p>
+         *
+         * @param stream the stream
+         * @param frame the frame
+         */
+        default void onStopSending(Stream stream, StopSendingFrame frame)
+        {
+        }
+
+        /**
+         * <p>Invoked when a RESET_STREAM frame has been received.</p>
+         * <p>This event is only emitted for informational purposes.</p>
+         *
+         * @param stream the stream
+         * @param frame the frame
+         */
+        default void onReset(Stream stream, ResetFrame frame)
+        {
+        }
+
+        /**
+         * <p>Invoked when the stream has been {@link Stream#isClosed() closed}.</p>
+         *
+         * @param stream the stream
+         */
+        default void onClose(Stream stream)
+        {
+        }
+
+        /**
+         * <p>Invoked when the stream is idle for longer than the idle timeout.</p>
+         *
+         * @param stream the stream
+         * @param failure the idle timeout failure
+         * @return whether to notify the other peer that this stream is closing
+         */
+        default boolean onIdleTimeout(Stream stream, TimeoutException failure)
+        {
+            return true;
+        }
+
+        /**
+         * <p>Invoked when a stream failure is detected.</p>
+         *
+         * @param stream the stream
+         * @param failure the stream failure
+         */
+        default void onFailure(Stream stream, Throwable failure)
+        {
+        }
+    }
+
+    /**
+     * <p>A {@link Retainable} wrapper for stream data bytes.</p>
+     */
+    interface Data extends Retainable
```

Comment:
Why isn't this a RBB in the same way that `Content.Chunk` is now?  It is-a `Retainable` and has-a `ByteBuffer`, so it is-a `RetainableByteBuffer` and can benefit from all the extra methods and utilities that we have for RBBs

Reply:
Removed, replaced by `Chunk`.